common/src/main/java/org/cloudfoundry/identity/uaa/authorization/UaaAuthorizationEndpoint.java

```diff
@@ -417,8 +429,8 @@ private String getUnsuccessfulRedirect(AuthorizationRequest authorizationRequest
             throw new UnapprovedClientAuthenticationException("Authorization failure, and no redirect URI.", failure);
         }
 
-        UriComponentsBuilder template = UriComponentsBuilder.fromUriString(authorizationRequest.getRedirectUri());
-        Map<String, String> query = new LinkedHashMap<String, String>();
+        UriComponentsBuilder template = uaaUrlUtils.parseAndDecodeUrl(authorizationRequest.getRedirectUri());
+        Map<String, String> query = new LinkedHashMap<>();
```

Comment:
Very little logic changes here. However, testing of this becomes a bit more complex. See below